src/program/frame.rs

```diff
@@ -77,6 +77,14 @@ impl FrameSet {
                 .map(|c| self.get_matching_keys(c))
                 .reduce(|acc, el| acc.into_iter().filter(|&v| el.contains(v)).collect())
                 .unwrap_or_default(),
+            FrameMatchCondition::Or(conditions) => conditions
+                .into_iter()
+                .map(|c| self.get_matching_keys(c))
+                .reduce(|mut acc, el| {
+                    acc.extend(el);
+                    acc
+                })
+                .unwrap_or_default(),
```

Comment:
I think this can be rewritten as

```suggestion
                .flatten()
                .collect(),
```